### PR TITLE
Moar optimizations

### PR DESCRIPTION
**Pull request by TheFakeElon**
## About The Pull Request

Optimizes /atom/movable/Move, might move NSV's bounding function to overmap movement only in another PR

Optimizes a few other things, mostly involving process(), overlays and overmaps

## Why It's Good For The Game

You shouldn't be able to roleplay as matrix neo in every gunfight with atmos enabled ;-;

## Changelog
:cl:
code: Optimizes misc process functions
code: Optimizes some overlay handling
code: Optimizes atom move
/:cl:

### CODE REVIEW

**Comment by athena413** [comment]
fix man good

File: nsv13/code/modules/overmap/overmap.dm [hunk 1]
```diff
@@ -621,34 +621,29 @@ Proc to spool up a new Z-level for a player ship and assign it a treadmill.
...
-			var/mob/mob = X
-			if(sound)
-				if(channel) //Doing this forbids overlapping of sounds
-					SEND_SOUND(mob, sound(sound, repeat = loop, wait = 0, volume = 100, channel = channel))
-				else
-					SEND_SOUND(mob, sound(sound, repeat = loop, wait = 0, volume = 100))
-			if(message)
-				to_chat(mob, message)
+/obj/structure/overmap/proc/relay(S, var/message=null, loop = FALSE, channel = null) //Sends a sound + text message to the crew of a ship
+	for(var/mob/M in mobs_in_ship)
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/overmap.dm:625]: Does this also need an as() ?

File: nsv13/code/__HELPERS/overmap.dm [hunk 2]
```diff
@@ -12,12 +12,10 @@ Helper method to get what ship an observer belongs to for stuff like parallax.
 */
 
 /mob/proc/find_overmap()
-	var/obj/structure/overmap/OM = loc?.get_overmap() //Accounts for things like fighters and for being in nullspace because having no loc is bad.
-	if(!OM) //We're on the overmap Z-level itself, thus we don't belong to any ship
-		if(last_overmap)
-			last_overmap.mobs_in_ship -= src
+	var/obj/structure/overmap/OM = loc.get_overmap() //Accounts for things like fighters and for being in nullspace because having no loc is bad.
```
Comment (covertcorvid) [inline nsv13/code/__HELPERS/overmap.dm:15]: Can loc ever be null here? I know it's uncommon but...

File: nsv13/code/modules/overmap/overmap.dm [hunk 3]
```diff
@@ -621,34 +621,29 @@ Proc to spool up a new Z-level for a player ship and assign it a treadmill.
...
+			SEND_SOUND(M, sound(S, repeat = loop, wait = 0, volume = 100))
+		if(message)
+			to_chat(M, message)
 
 /obj/structure/overmap/proc/stop_relay(channel) //Stops all playing sounds for crewmen on N channel.
-	for(var/X in mobs_in_ship)
-		if(ismob(X))
-			var/mob/mob = X
-			mob.stop_sound_channel(channel)
+	for(var/mob/M in mobs_in_ship)
```
Comment (covertcorvid) [inline nsv13/code/modules/overmap/overmap.dm:634]: or this one?

File: nsv13/code/__HELPERS/overmap.dm (hunk 2 above)
Comment (TheFakeElon) [reply nsv13/code/__HELPERS/overmap.dm:15]: if loc is null, something must have already broke catastrophically. It's not worth the (admittedly small) performance dent to check if it exists every time a mob moves

**Review by Bokkiewokkie — APPROVED** [review]

**Review by covertcorvid — APPROVED** [review]
Haven't noticed any problems

**Pull request merged**